crates/nu-engine/src/compile/mod.rs

```diff
@@ -194,11 +194,24 @@ fn compile_pipeline(
             out_reg,
         )?;
 
-        // Clean up the redirection
-        finish_redirection(builder, redirect_modes, out_reg)?;
+        // only clean up the redirection if current element is not
+        // a subexpression.  The subexpression itself already clean it.
+        if !is_subexpression(&element.expr.expr) {
+            // Clean up the redirection
+            finish_redirection(builder, redirect_modes, out_reg)?;
+        }
 
         // The next pipeline element takes input from this output
         in_reg = Some(out_reg);
     }
     Ok(())
 }
+
+fn is_subexpression(expr: &Expr) -> bool {
+    match expr {
+        Expr::FullCellPath(inner) => {
+            matches!(&inner.head.expr, &Expr::Subexpression(..))
```

Comment:
I am certainly missing some context, but why is this covering just the case of a subexpression as part of a `FullCellPath`? Wouldn't we also (or primarily) need to cover the case of `Subexpression` itself?

Reply:
It's because all subexpression exists with part of `FullCellPath` firstly.  So `Expr::Subexpression` will never exists in `compile_pipeline` function.
Maybe it's still ok to cover the case of Subexpression, but I don't think it's necessary.

Reply:
I would add it. If someone refactors the parsing there could be the reasonable result that we don't have unnecessary `FullCellPaths` wrapping the subexpressions anymore. (its not really necessary if there is no `.some_col` following, right)

Reply:
Make sence to me.  Just added this.